engines/ic3ia.cpp

```diff
@@ -186,7 +185,9 @@ void IC3IA::abstract()
   // add predicates automatically added by ia_
   // to our predset_
   // needed to prevent adding duplicate predicates later
-  predset_.insert(bool_symbols.begin(), bool_symbols.end());
+  for (const auto & sym : bool_symbols) {
+    add_predicate(sym);
```

Comment:
why we are adding the boolean symbols? previously we were just adding them in predset_

Reply:
Right, good question. So I was worried about the tight integration with `ceg-prophecy-arrays`. With this change, the refinement step needs to not only add predicates back in but also add boolean symbols to both `predset_` and `predlbls_` or else there will be issues in `IC3IA`. I thought that was annoying and complicated so I just wrapped it all into `add_predicate` which checks if it's a symbol (or a literal more specifically) and does the right thing if so.

Reply:
I see. Then, are you doing this change in Ceg-proph (in the refine-subprover-ts)?

Reply:
Oops good point. Fixed with 4ccf3cde6e75b562306cb712d30a5743425862f7.

I cleared `predlbls_` in `refine-subprover-ts` in a previous commit.